### ISSUE TITLE
Should throw on `async () => await 5 ** 6;`

### ISSUE THREAD

**Issue by fisker** [eslint/espree#472]
`espree` parse this as `await (5 ** 6)`

```text
require('espree').parse('async () => await 5 ** 6', {ecmaVersion: 2017}).body[0].expression.body
Node {
  type: 'AwaitExpression',
  start: 12,
  end: 24,
  argument: Node {
    type: 'BinaryExpression',
    start: 18,
    end: 24,
    left: Node { type: 'Literal', start: 18, end: 19, value: 5, raw: '5' },
    operator: '**',
    right: Node { type: 'Literal', start: 23, end: 24, value: 6, raw: '6' }
  }
}
````

In chrome, this is a syntax error

```js
async () => await 5 ** 6
VM40:1 Uncaught SyntaxError: Unary operator used immediately before exponentiation expression. Parenthesis must be used to disambiguate operator precedence
```

Babel PR babel/babel#12441

**Comment by nzakas** [comment]
Can you please provide details in this issue? A link to a PR isn’t enough for us to evaluate this.

**Comment by fisker** [comment]
I updated the issue description.

**Comment by nzakas** [comment]
Thanks. This is actually an issue with Acorn, which Espree wraps. I’d suggest opening an issue there. We can pull in the updated version when it’s fixed.

**Comment by fisker** [comment]
I think acorn already fixed it https://github.com/acornjs/acorn/commit/848d14e20653a211b06c8e167810bd4fae2173cf

**Comment by nzakas** [comment]
Awesome, then we can look at incorporating it.

**Comment by mdjermanovic** [comment]
@fisker it seems that the latest Acorn still doesn't throw an error on this particular example. Can you open an issue on https://github.com/acornjs/acorn repository?

**Comment by robpalme** [comment]
@mdjermanovic as you requested, I have opened [an issue](https://github.com/acornjs/acorn/issues/1028) and [created a PR](https://github.com/acornjs/acorn/pull/1029) to fix this.

**Comment by fisker** [comment]
Somehow, I missed this. Thank you @robpalme!

**Comment by mdjermanovic** [comment]
@robpalme thanks!

**Comment by robpalme** [comment]
This is now fixed in the Acorn 8.2.2 release.

**Comment by nzakas** [comment]
You are my hero @robpalme. Thank you.
